### DISCUSSION TITLE
How to actually install FluidSynth under 10 Enterprise 64bit?

### DISCUSSION THREAD

**Discussion by The-Great-Sephiroth** [FluidSynth/fluidsynth#1341] [General]
I'm sorry if I am missing something basic, but I cannot for the life of me find any information as to how to use FluidSynth on Windows. I simply want it to be my default MIDI device for playback, not recording. I downloaded the binary, extracted the BIN folder, copied it to C:\Program Files\FluidSynth, and tried running the executable. This threw an error about the default soundfont not existing. I have a 144MiB soundfont ready to go, but cannot figure out how to configure FS or how to make it show up as a MIDI device in 10. There was no installer or anything. No configuration. Nothing. I am literally stumped.

One other thing. This is a laptop. A modern laptop. When running the executable it complains about not finding a MIDI device. Well, no duh! This is a modern system. Systems have not had MIDI devices in twenty years! I want FS to be my MIDI device so i can actually use MIDI again. What am I missing here?

**Comment by derselbst** [comment]
I suggest you read the docs: https://github.com/FluidSynth/fluidsynth/wiki/GettingStarted

> Systems have not had MIDI devices in twenty years!

Great, so how about plugging in a MIDI device?

>  I want FS to be my MIDI device so i can actually use MIDI again.

Fluidsynth is **no** MIDI device. It's a software synthesizer, that _connects_ to MIDI devices. It's not clear to me what you're trying to do...

**Reply by The-Great-Sephiroth** [reply]
I was told that FluidSynth replaced Timidity++. Timidity could be used to play MIDI and was used heavily in emulation software for DOS, like DOSBox or pcem. I assumed that since this replaced it, I could use it. DOSBox-X even has configuration for FluidSynth. In fact as a test I set it up to use FS, but cannot figure out what to do with the files in "bin" to make it all work. A simple installer would likely have solved all of my issues, or something telling me what I need to do to make it work. All I can find is source code and binary blobs though.

I basically want MIDI playback on the level that my Ensoniq Audio PCI card allowed for. Back then I only had an 8MB waveset. I would like to use my old soundfonts as well as new ones to enjoy my massive collection of MIDI from the 90's as well as to use with DOSBox-X or other DOS software when I am feeling nostalgic.

Are you saying that FluidSynth is incapable of MIDI playback on a PC with a modern soundcard for things like this?

*EDIT*

I start it with the -n parameter and it stops fussing about MIDI devices.

**Reply by The-Great-Sephiroth** [reply]
Okay, I think I have it partially figured out. If I open a DOS prompt and start it with "-n -s" to run as a server and disable input (no MID ports for input) I can then use it flawlessly inside of DOSBox-X. Sounds amazing.

With that solved, how would I tell Media Player or whatever to use FS instead of the horrible Windows software MIDI? My train of thought was that I could somehow "install" this software and select it for MIDI playback in place of the default one. I would also assume I could use it without manually starting it at every boot and not have a DOS window open the entire time.

**Reply by derselbst** [reply]
Fluidsynth is a commandline application. You just open a commandline, cd to the bin folder and execute it as pointed out in the docs. Eventually adding the bin folder to your PATH for convenience. I'd assume that people who have programmed in C, C++, Pascal before are familiar with this procedure. If you want an installer for that simple task, you're welcome to contribute one - no one has done it so far.

If you're looking for a graphical MIDI player, the documentation lists various applications that use fluidsynth. Yet, fluidsynth cannot be integrated into Media Player.

I am aware that the DOS community uses fluidsynth. But that's all I know, so you better ask them.

**Reply by The-Great-Sephiroth** [reply]
I know all of what you said. I was trying to run it without the window. I can easily make a shortcut or something, but it still opens its own window.

What I am looking for is something that I can install, manually or with an installer, that will be a selectable MIDI playback device, like the "Microsoft GS Wavetable" or whatever it is, but that does a job as good as FS. FS is working and I did try a game or two with it and it works wonderfully, but as you stated, Media Player and other devices are not able to "select" it for playback and only show the MS option.

For now I am okay with it and I believe I have it understood well enough for my DOS stuff. I just need to open it when starting DOS, and close it when done.
